cert-generation/main.go

```diff
@@ -0,0 +1,127 @@
+	webhookServerName = "webhook-server"
+)
+
+func main() {
+	log.SetOutput(os.Stdout)
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatal("Failed when attempting to retrieve in cluster config: ", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatal("Failed when attempting to retrieve in cluster config: ", err)
+	}
+
+	namespaceByte, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		log.Fatal("Could not retrieve namespace: ", err)
+	}
+
+	namespace := string(namespaceByte)
+	configMapData := make(map[string]string, 0)
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      certConfigMapName,
+			Namespace: namespace,
+			Annotations: map[string]string{
+				"service.beta.openshift.io/inject-cabundle": "true",
+			},
+		},
+		Data: configMapData,
+	}
+
+	// Create the configmap or update if it already exists
+	if _, err := clientset.CoreV1().ConfigMaps(namespace).Get(certConfigMapName, metav1.GetOptions{}); errors.IsNotFound(err) {
+		_, err = clientset.CoreV1().ConfigMaps(namespace).Create(configMap)
+		if err != nil {
+			log.Fatal("Failed when attempting to create configmap: ", err)
+		}
+	} else {
+		_, err = clientset.CoreV1().ConfigMaps(namespace).Update(configMap)
+		if err != nil {
+			log.Fatal("Failed when attempting to update configmap: ", err)
+		}
+	}
+
+	label := map[string]string{"app": "che-workspace-controller"}
+
+	port := int32(443)
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secureServiceName,
+			Namespace: namespace,
+			Labels:    label,
+			Annotations: map[string]string{
+				"service.beta.openshift.io/serving-cert-secret-name": certSecretName,
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Port:       port,
+					Protocol:   "TCP",
+					TargetPort: intstr.FromString(webhookServerName),
+				},
+			},
+			Selector: label,
+		},
+	}
+
+	// Create secure service or update it if it already exists
+	if clusterService, err := clientset.CoreV1().Services(namespace).Get(secureServiceName, metav1.GetOptions{}); errors.IsNotFound(err) {
+		_, err = clientset.CoreV1().Services(namespace).Create(service)
+		if err != nil {
+			log.Fatal("Failed when attempting to create service: ", err)
+		}
+	} else {
+		// Cannot naively copy spec, as clusterIP is unmodifiable
+		clusterIP := clusterService.Spec.ClusterIP
+		service.Spec = clusterService.Spec
+		service.Spec.ClusterIP = clusterIP
+		service.ResourceVersion = clusterService.ResourceVersion
+
+		_, err = clientset.CoreV1().Services(namespace).Update(service)
+		if err != nil {
+			log.Fatal("Failed when attempting to update service: ", err)
+		}
+	}
+
+	// Wait until it's cleaned up by the workspace-controller
+	log.Println("I've done my work and waiting to be utilized. Don't feel sorry for me I was initially designed to die if everything is fine")
```

Comment:
If was a joke but up to you to leave it or reformulate )

Reply:
I thought it was a funny secret haha